tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/strategy/ProgrammeBasedFilterStrategy.java

```diff
@@ -3,33 +3,80 @@
 import com.transformuk.hee.tis.security.model.Programme;
 import com.transformuk.hee.tis.security.model.UserProfile;
 import com.transformuk.hee.tis.security.util.TisSecurityHelper;
+import com.transformuk.hee.tis.tcs.api.enumeration.ProgrammeMembershipStatus;
+import com.transformuk.hee.tis.tcs.service.model.ColumnFilter;
+import java.util.List;
 import java.util.Optional;
 import java.util.Set;
+import java.util.stream.Collectors;
 import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.lucene.search.join.ScoreMode;
 import org.elasticsearch.common.collect.Tuple;
 import org.elasticsearch.index.query.BoolQueryBuilder;
 import org.elasticsearch.index.query.MatchQueryBuilder;
 import org.elasticsearch.index.query.NestedQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
 import org.springframework.stereotype.Component;
 
 @Component
 public class ProgrammeBasedFilterStrategy implements RoleBasedFilterStrategy {
 
-  private static final String COLUMN_FILTER = "programmeName";
+  private static final String COLUMN_FILTER = "programmeId";
 
   @Override
-  public Optional<Tuple<String, BoolQueryBuilder>> getFilter() {
+  public Optional<Tuple<String, BoolQueryBuilder>> getFilter(List<ColumnFilter> columnFilters) {
+    MatchQueryBuilder statusMatchQueryBuilder = null;
```

Comment:
It looks like the `statusMatchQueryBuilder` is only used `if (CollectionUtils.isNotEmpty(assignedProgrammes))` so could sit within that block.

Reply:
Done.